akka-sample-kafka-to-sharding-scala/processor/src/main/scala/sample/sharding/kafka/UserEventsKafkaProcessor.scala

```diff
@@ -29,35 +29,42 @@ object UserEventsKafkaProcessor {
         implicit val ec: ExecutionContextExecutor = ctx.executionContext
         implicit val scheduler: Scheduler = classic.scheduler
 
+        // ActorContext can't be accessed inside Futures
+        // so we capture the scheduler and log before using them
+        val typedScheduler = ctx.system.scheduler
+        val logger = ctx.log
```

Comment:
This is against the intended usage of `ctx.log`. When calling `ctx.log` it initializes mdc stuff and that is also cleared after message processing.